backend/fms_core/template_importer/sheet_data.py

```diff
@@ -67,4 +74,16 @@ def generate_preview_info_from_rows_results(self, rows_results):
             "valid": self.is_valid,
             "base_errors": self.base_errors,
             "rows": rows_results,
-        }
\ No newline at end of file
+        }
+    
+    def check_for_empty_data_array(self,row_data):
+        for x in (row_data):
+            if x:
+                return False
+        return True
+    
+    def check_for_empty_row_errors(self):
```

Comment:
this does not deserve a function this won't be reused. It is just less readable you have to actually fetch the function to know it just raise an error... you can